osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs

```diff
@@ -187,25 +232,219 @@ private void updateState()
             });
         }
 
-        private void updateButtonColour(bool green)
+        public class ReadyButton : Components.ReadyButton
```

Comment:
Definitely needs a better name. `MultiplayerReadyButton.ReadyButton : Components.ReadyButton` is nightmarish.

Also I feel this file has hit the imaginary LoC limit for a single class, it's very hard to read through. Maybe can be split out further rather than subclassed.

Reply:
I agree. But I think that should be done in a separate pass.
- The outer class shouldn't be called a "button".
- The inner classes should be split out into separate files.

Reply:
At the point the diff here is not useful, is there any reason to do it as a separate pass?

I think the control can be called `MatchStartControl` or something like that.